### PR TITLE
feat: Update docs for Konnect support of vault secret TTL rotation

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Description

What did you change and why?
- Added release note that explains the new feature
- Modified the Secrets/Vault overview bullet a little
- Added a new step to the vault config instructions as well as links to Gateway vault config docs
 
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.
DOCU-3351

### Testing instructions

Netlify link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/konnect/runtime-manager/configuration/vaults/how-to.md [hunk 1]
```diff
@@ -20,6 +20,10 @@ Open the **Vaults** page in {{site.konnect_short_name}}:
 Set up a new vault. For this example, we're going to use the environment variable vault. 
 1. Click **Add vault**.
 1. Choose a vault type. 
+1. Enter the configuration settings for your vault. For more information about how to configure settings, see the following {{site.base_gateway}} documation:
+    * [AWS vault configuration options](/gateway/latest/kong-enterprise/secrets-management/backends/aws-sm/#vault-configuration-options)
+    * [Google Cloud vault configuration options](/gateway/latest/kong-enterprise/secrets-management/backends/gcp-sm/#vault-entity-configuration-options)
+    * [HashiCorp vault configuration options](/gateway/latest/kong-enterprise/secrets-management/backends/hashicorp-vault/#vault-configuration-options)
```
Comment (cloudjumpercat) [inline app/konnect/runtime-manager/configuration/vaults/how-to.md:26]: ❗ Reviewers: Do you like the links to Gateway docs? I feel like the UI isn't helpful in telling you what you need to put there or what they do, so I felt like we needed to include something. This is kind of an "MVP" solution, I think maybe the Konnect docs need their own table? Or maybe we just continue linking to Gateway to point to how the two interact?

File: app/konnect/runtime-manager/configuration/vaults/how-to.md [hunk 2]
```diff
@@ -20,6 +20,10 @@ Open the **Vaults** page in {{site.konnect_short_name}}:
 Set up a new vault. For this example, we're going to use the environment variable vault. 
 1. Click **Add vault**.
 1. Choose a vault type. 
+1. Enter the configuration settings for your vault. For more information about how to configure settings, see the following {{site.base_gateway}} documation:
```
Comment (lena-larionova) [inline app/konnect/runtime-manager/configuration/vaults/how-to.md:23]:
```suggestion
1. Enter the configuration settings for your vault. For more information about how to configure settings, see the following {{site.base_gateway}} documentation:
```
How did Vale not catch this 🤔

File: app/konnect/runtime-manager/configuration/vaults/how-to.md (hunk 1 above)
Comment (lena-larionova) [reply app/konnect/runtime-manager/configuration/vaults/how-to.md:26]: The links work for me. I don't think we should worry about this too much on our end; if anything, we should be pointing out areas for UI improvements.

**Comment by cloudjumpercat** [comment]
@lena-larionova When you get a chance, could you approve this? Thanks!

**Review by lena-larionova — APPROVED** [review]
LGTM!

**Comment by cloudjumpercat** [comment]
Ignoring Vale failures, they are on other docs and I fixed/resolved all the ones on this PR.

**Comment by smritikjaggi** [comment]
Publishing as 3.4 is live in Konnect

**Pull request merged**